anime_tracker/src/models/APIHandler.ts

```diff
@@ -0,0 +1,13 @@
+import {  OverviewArray } from "./animeSearch";
+
+export class APIHandler {
+  APICALL: string = "https://api.jikan.moe/v4/";
+
+  getAnimeSearch = async (value: string) => {
+    const response = await fetch(this.APICALL+`anime?q=${(value)}`);
+    if (!response.ok) {  console.error(response); }
```

Comment:
I recommend creating a handler for responses that you can use to handle failed responses

Reply:
const response = await fetch(**this.APICALL+\`anime?q=${(value)}\`**);

either use template literals or simple strings, but avoid combining them togheter

Reply:
```
export class APIHandler {
  //basic API request variables
  base_url: string = "https://api.jikan.moe/v4/";
  getAnime_extension: string = "anime/";
  question_extension: string = "?q=";

  //get anime by name
  getAnimeSearch = async (value: string) => {
    const response = await fetch(this.base_url+this.getAnime_extension+this.question_extension+`${(value)}`);
    if (!response.ok) {  console.error(response); }
    const data = await response.json();
    return data.data as Overview;
  };
}
```

like that?